code/game/objects/items/RCD.dm

```diff
@@ -1133,62 +1164,117 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
 	return ..()
 
 /obj/item/construction/plumbing/attack_self(mob/user)
-	..()
-	if(!choices.len)
-		for(var/obj/machinery/plumbing/plumbing_type as anything in plumbing_design_types)
-			choices += list(initial(plumbing_type.name) = image(initial(plumbing_type.icon), icon_state = initial(plumbing_type.icon_state)))
-			name_to_type[initial(plumbing_type.name)] = plumbing_type
-			machinery_data["cost"][plumbing_type] = plumbing_design_types[plumbing_type]
+	ui_interact(user)
 
-	// Update duct icon
-	var/image/duct_image = choices["fluid duct"]
-	duct_image.color = current_color
+/obj/item/construction/plumbing/examine(mob/user)
+	. = ..()
+	. += "You can scroll your mouse wheel to change the piping layer."
+	. += "You can right click a fluid duct to set the Plumbing RPD to its color and layer."
+
+/obj/item/construction/plumbing/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "PlumbingService", name)
+		ui.open()
+
+/obj/item/construction/plumbing/ui_assets(mob/user)
+	return list(
+		get_asset_datum(/datum/asset/spritesheet/plumbing),
+	)
 
-	var/choice = show_radial_menu(user, src, choices, custom_check = CALLBACK(src, PROC_REF(check_menu), user), require_near = TRUE, tooltips = TRUE)
-	if(!choice || !check_menu(user))
+/obj/item/construction/plumbing/ui_static_data(mob/user)
+	return list("paint_colors" = GLOB.pipe_paint_colors)
+
+/obj/item/construction/plumbing/ui_data(mob/user)
+	/**
+	 * Maps current_layer[type string] -> number so it can be sent to the UI
+	 * I didn't want to create an temporary list to map these values as it would take up memory so i made this dumb logic
+	 * GLOB.plumbing_layers maps layer name["First Layer","Second Layer","Default Layer","Forth Layer","Fifth Layer"]  to bit mask values 1,2,4,8,16
+	 * We need to map these values 1,2,4,8,16 -> 1,2,3,4,5 i.e list index values
```

Comment:
Please please just hard code this instead, then you won't need this comment or the confusing code

Reply:
This whole comment is longer than the code that you would need

Reply:
I mean it works why change it? Fine i'll just hard code the values

Reply:
Because it's hard to read and hard to understand and isn't guaranteed to work if we change the values